docs/guides/typescript.md

```diff
@@ -158,42 +158,30 @@ The typestates of a machine are specified as the 3rd generic type in `createMach
 **Example:**
 
 ```ts
-import { createMachine, interpret } from 'xstate';
+import {
+  CombineTypestates,
+  TypestateContext,
+  createMachine,
+  interpret
+} from 'xstate';
 
 interface User {
   name: string;
 }
 
-interface UserContext {
-  user?: User;
-  error?: string;
-}
-
 type UserEvent =
   | { type: 'FETCH'; id: string }
   | { type: 'RESOLVE'; user: User }
   | { type: 'REJECT'; error: string };
 
-type UserState =
-  | {
-      value: 'idle';
-      context: UserContext & {
-        user: undefined;
-        error: undefined;
-      };
-    }
-  | {
-      value: 'loading';
-      context: UserContext;
-    }
-  | {
-      value: 'success';
-      context: UserContext & { user: User; error: undefined };
-    }
-  | {
-      value: 'failure';
-      context: UserContext & { user: undefined; error: string };
-    };
+type UserState = CombineTypestates<
```

Comment:
Can you add a `/** typedoc */` explaining what this does (presumably combines `context` and `value` for each of the typestates)?